pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDMintingPipeline.java

```diff
@@ -0,0 +1,247 @@
+        for (String uniqueTerm : uniqueTerms){
+            Optional<DwcTerm> dwcTerm = getDwcTerm(uniqueTerm);
+            if(dwcTerm.isPresent()){
+                uniqueDwcTerms.add(dwcTerm.get());
+            } else {
+                throw new RuntimeException("Unrecognised unique term configured for datasource " + options.getDatasetId() + ", term: " + uniqueTerm);
+            }
+        }
+
+        log.info("Transform 1: ExtendedRecord er ->  <uniqueKey, er.getId()> - this generates the UniqueKey.....");
+        PCollection<KV<String, String>> extendedRecords =
+                p.apply(AvroIO.read(ExtendedRecord.class).from(options.getTargetPath() + "/" + options.getDatasetId().trim() + "/1/interpreted/verbatim/*.avro"))
+                .apply(ParDo.of(new DoFn<ExtendedRecord, KV<String, String>>() {
+                    @ProcessElement
+                    public void processElement(@Element ExtendedRecord source, OutputReceiver<KV<String, String>> out, ProcessContext c) {
+                        out.output(KV.of(generateUniqueKey(source, uniqueDwcTerms), source.getId()));
+                    }
+        }));
+
+        PCollection<KV<String, String>> alaUuids = null;
+
+        log.info("Transform 2: ALAUUIDRecord ur ->  <uniqueKey, uuid> (assume incomplete)");
+        File alaRecordDirectory = new File(alaRecordDirectoryPath);
+
+        if (alaRecordDirectory.exists() && alaRecordDirectory.isDirectory()){
+            alaUuids = p.apply(AvroIO.read(ALAUUIDRecord.class)
+                            .from(alaRecordDirectory + "/*.avro"))
+                            .apply(ParDo.of(new ALAUUIDRecordKVFcn()));
+        } else {
+
+            TypeDescriptor<KV<String,String>> td = new TypeDescriptor<KV<String,String>>() {};
+            log.warn("[WARNING] Previous ALAUUIDRecord records where not found. This is expected for new datasets, but is a problem" +
+                    "for previously loaded datasets - will mint new ones......");
+            alaUuids = p.apply(Create.empty(td));
+        }
+
+        log.info("Create join collection");
+        PCollection<KV<String, KV<String, String>>> joinedPcollection =
+                org.apache.beam.sdk.extensions.joinlibrary.Join.fullOuterJoin(extendedRecords, alaUuids, NO_ID_MARKER, NO_ID_MARKER);
+
+        log.info("Create ALAUUIDRecords and write out to AVRO");
+        joinedPcollection
+                .apply(ParDo.of(new CreateALAUUIDRecordFcn()))
+                .apply(AvroIO.write(ALAUUIDRecord.class)
+                        .to(alaRecordDirectory + "_new/interpret")
+                        .withSuffix(".avro")
+                        .withCodec(BASE_CODEC));
+
+        log.info("Running the pipeline");
+        PipelineResult result = p.run();
+        result.waitUntilFinish();
+        MetricsHandler.saveCountersToTargetPathFile(options, result.metrics());
+
+        //TODO duplicate check of ALAUUIDRecord entries and report them.....
```

Comment:
If this is being skipped in this PR, there should be an issue link here to make sure that we can track it being implemented in the future.